api/envoy/service/auth/v3/attribute_context.proto

```diff
@@ -116,7 +117,16 @@ message AttributeContext {
     // The HTTP request headers. If multiple headers share the same key, they
     // must be merged according to the HTTP spec. All header keys must be
     // lower-cased, because HTTP header keys are case-insensitive.
-    map<string, string> headers = 3;
+    // Header value is encoded as string. This does not work for non-utf8 characters.
+    map<string, string> headers = 3
```

Comment:
When  header_as_bytes is false, we have to encode the header here for backwards compatibility.  Otherwise, we can directly encode in the HeaderValue:value field below: https://github.com/envoyproxy/envoy/blob/372a262894f028431cb84f3a1d361fb8fb6a2518/api/envoy/config/core/v3/base.proto#L331

This makes the proto a little bit redundant, but is a price we need to pay for backwards compatibility.

Reply:
If headers_as_bytes is true, we can't use HeaderValue:value because it's a string field.

In this PR I am using HeaderValue:raw_value, which is a bytes field, if headers_as_bytes is true. I have not changed the default behavior of when headers_as_bytes is false.